share/completions/go.fish

```diff
@@ -2,33 +2,43 @@
 # See: https://golang.org
 
 function __fish__go_buildflags_completions -d 'Complete go build commands' --argument-names cmd
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -s a -d 'force rebuild'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -s n -d 'print the commands but do not run them'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -s p -r -d 'number parallel builds (default=#cpus)'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o race -d 'enable data race detection'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -s v -d 'print packages being built'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o work -d 'print and preserve work directory'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -s x -d 'print the commands'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o ccflags -r -d 'c compiler flags'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o compiler -x -d 'name of compiler to use, as in runtime' -a 'gccgo gc'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o gccgoflags -r -d 'gccgo compiler/linker flags'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o gcflags -r -d 'go compiler flags'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o installsuffix -r -d 'suffix for installation directory'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o ldflags -r -d 'linker flags'
-        complete -c go -n "__fish_seen_subcommand_from $cmd" -o tags -r -d 'build tags'
+    complete -c go -n "__fish_seen_subcommand_from $cmd" -s a -d 'force rebuild'
```

Comment:
`__fish_seen_subcommand_from` takes multiple arguments, and returns true if one of them was seen - i.e. it's meant like "seen a subcommand from this list of subcommands".

That means instead of making this function and calling it in multiple places, you can just do

```fish
complete -c go -n "__fish_seen_subcommand_from $list_of_commands"
```

e.g.

```fish
complete -c go -n "__fish_seen_subcommand_from build doc install [...]" -x -a "(
                __fish_print_go_packages (commandline -ct)
        )" --description 'Package'
```